### ISSUE TITLE
Message provider naming does not correspond to the conventions

### ISSUE THREAD

**Issue by Kathrin84** [academic-moodle-cooperation/moodle-mod_organizer#24]
Hi together,

a user told us that a message provider of the organizer uses a sign ( : ) that is not allowed to use there:
- message_provider_mod_organizer_appointment_reminder**:**student_loggedin is used
- message_provider_mod_organizer_appointment_reminder**_**student_loggedin should be used

Allowed is: [A-Za-z0-9_-]

Some code points where to find the providers:
- https://github.com/academic-moodle-cooperation/moodle-mod_organizer/blob/be339319d2e8e93597607100f91d0328a090862f/lib.php#L813
- https://github.com/academic-moodle-cooperation/moodle-mod_organizer/blob/c4a66d51efbb878978deaf852fd3012b7e9cc4cf/db/messages.php#L36

This should be a small fix and maybe you plan to integrate it into a new release version.

Cheers, Kathrin

**Comment by mkemmerling** [comment]
Thanks for reporting. I forwarded the bug to the developer.

**Comment by Kathrin84** [comment]
Hi Markus,
I just wanted to ask what's the state of this issue as this bug is still existing and it prevents the user from changing his notification preferences because this error is shown and his user preference cannot be saved for your plugin:
```
preferences => Ungültiger Parameterwert gefunden: type => Ungültiger Parameterwert gefunden: Invalid external api parameter: the value is "message_provider_mod_organizer_slotdeleted_notify:student_loggedin", the server was expecting "alphanumext" type
Error code: invalidparameter
* line 333 of /lib/externallib.php: invalid_parameter_exception thrown
* line 224 of /lib/externallib.php: call to external_api::validate_parameters()
* line 59 of /lib/ajax/service.php: call to external_api::call_external_function()
```
Best, Kathrin

**Comment by mkemmerling** [comment]
Hi Kathrin,
what Moodle version are you currently on? The bug should not longer occur in the Organizer version for Moodle 3.3.

**Comment by Kathrin84** [comment]
Hi Markus,
we're updating to 3.2 tomorrow - so it's 3.2.
But after your comment with 3.3 I've tested it there and it does not throw an error anymore. So I assume it's fixed for this version. Then you can close this issue with the hint to the fix version and I have to discuss with my colleague what we do for our 3.2 version.
Best, Kathrin

**Comment by mkemmerling** [comment]
Hi Kathrin,
I am sorry, but we won't have the time to backport the fix to 3.2. The changes were a little bit too involved for a quick fix.
